include/libopencm3/nrf5/2/memorymap.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2018
+ * Luong Le <novercy@live.com>
+ * Gerrit Maus <funk@maus.xyz>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef LIBOPENCM3_MEMORYMAP_H
+#define LIBOPENCM3_MEMORYMAP_H
+
+#include <libopencm3/cm3/memorymap.h>
+
+/* --- Convenience defines ----------------------------------------------- */
+#define PTRIO32(addr)              ((volatile uint32_t *)(addr))
+#define PTRIO64(addr)              ((volatile uint64_t *)(addr))
+
+/* --- NRF52 specific peripheral definitions ----------------------------- */
+
+/* Memory map for all busses */
+#define PERIPH_BASE                0x40000000U
+#define PERIPH_BASE_APB            0x40000000U
+#define PERIPH_BASE_AHB            0x50000000U
+
+/* Register boundary addresses */
+
+/* APB */
+#define CLOCK_BASE                (PERIPH_BASE_APB + 0x0000)
```

Comment:
Are these really all meant to have the ~same address?

Reply:
Yes, some peripherals have the same address